src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +804,765 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
     return CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetThreadNetworkDiagnosticAttributeInfo(
+    chip::AttributeId attributeId, uint8_t ** buffer, uint16_t & ReadLength, chip::TLV::TLVType & type)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    switch (attributeId)
+    {
+    case ZCL_CHANNEL_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint16_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        uint16_t channel = static_cast<uint16_t>(otLinkGetChannel(mOTInst));
+        Encoding::LittleEndian::Put16(*buffer, channel);
+    }
+    break;
+
+    case ZCL_ROUTING_ROLE_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(otDeviceRole);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        otDeviceRole role = otThreadGetDeviceRole(mOTInst);
+        Encoding::LittleEndian::Put16(*buffer, role);
+    }
+    break;
+
+    case ZCL_NETWORK_NAME_ATTRIBUTE_ID: {
+        type       = kTLVType_UTF8String;
+        ReadLength = OT_NETWORK_NAME_MAX_SIZE + 1;
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        ChipLogProgress(DeviceLayer, "Read NetworkName= %s", otThreadGetNetworkName(mOTInst));
+        strncpy(reinterpret_cast<char *>(*buffer), otThreadGetNetworkName(mOTInst), ReadLength);
+        ChipLogProgress(DeviceLayer, "Copied NetworkName= %s", *buffer);
+    }
+    break;
+
+    case ZCL_DIAG_PAN_ID_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(otPanId);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        otPanId panId = otLinkGetPanId(mOTInst);
+        Encoding::LittleEndian::Put16(*buffer, panId);
+    }
+    break;
+
+    case ZCL_DIAG_EXTENDED_PAN_ID_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint64_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        const otExtendedPanId * pExtendedPanid = otThreadGetExtendedPanId(mOTInst);
+        Encoding::LittleEndian::Put64(*buffer, Encoding::BigEndian::Get64(pExtendedPanid->m8));
+    }
+    break;
+
+    case ZCL_MESH_LOCAL_PREFIX_ATTRIBUTE_ID: {
+        const uint8_t kOctStrPrefixLen =
```

Comment:
If `formattedIpv6Prefix` is declared first, can use `2*sizeof(formattedIpv6Prefix) + 1`, for clarity....